webui/react/src/stores/cluster.tsx

```diff
@@ -84,6 +84,9 @@ const clusterStatusText = (
   )}%`;
 };
 
+const updateIfChanged = <T, V extends WritableObservable<T>>(o: V, next: T) =>
```

Comment:
This looks like a pretty helpful utility function. Is there a good place for this to live and be shared?

Reply:
Could tag a `memoizedUpdate` method onto the prototype in `observable.ts`.

Reply:
Good point -- I'm sort of hesitant to put it in the observables utilities mainly because this pattern doesn't look like it's in use anywhere else and i'm not sure it's a pattern we want to spread.

Reply:
Could also just be
```typescript
const preserveIdentity = <T>(current: T, next: T) => isEqual(current, next) ? current : next)
```
so it's not observable specific.